SU-HSDO/suhumsci
# feat(STN101,STN83):  node detail page layout and news in horizontal card pattern

# READY FOR REVIEW

# Summary
Includes the changes from both https://github.com/SU-HSDO/suhumsci/pull/444 and https://github.com/SU-HSDO/suhumsci/pull/447

Please reference these PRs for comments and approving reviews if needed.

## Summary of 101
This brings the `three-column-w-image` layout over to the `humsci_basic` theme, and includes some basic layout styles that are applied to the page.

I used as much of the Decanter breakpoint system as I could, (everything but the largest breakpoint).

I opted for a custom flexbox layout over the Decanter one just because our designs used different sizes for the sidebar and gutters. Also, having the sidebars scale down the same rate as the main content wasn't working that well.

## Summary of 83

Creates the styles and HTML for the horizontal card and News Item content type. We are not worrying about events in this context yet.

Additionally, I created some SCSS functions and mixins for looking up the primary, secondary, and tertiary colors for the colorful-teal theme variation. Also, added sass-true tests.

Design: https://www.figma.com/file/toNZnFLadtpNpG5dPzNGWz/%E2%9A%99%EF%B8%8F-Colorful-Theme---Working?node-id=1408%3A1313

# Need Review By (Date)
asap

# Urgency
'high'

# Steps to Test STN 101
1. Browse to the page for a News item.
2. Verify that the single-column content is centered at large screen widths, and has screen margins at small screen width.
3. Using layout-builder, add some content to the left-sidebar, save the layout, and confirm that the "sidebar + main content" layout displays appropriately across all screen widths.
4. Using layout-builder, add some content to the right-sidebar, save the layout, and confirm that the "sidebar + main content" layout displays appropriately across all screen widths.

Note: content can be added to both sidebars though that use-case isn't optimized. TBD whether this is out of scope for this issue.

# Steps to Testing 83
1. Add a news test page to your local drupal:
- On an existing site (using the colorful theme), create a few pieces of dummy “News Item” content. Remember to add some without images, one without an external link, etc.
- Create an empty “basic page.”
- On the basic page, add a paragraph:
  - View → News
    - Display: → List
- Ensure that the news items are being displayed on the page and the default looks good
(Note that layout is not created yet, so everything will be 100%)
2. inside humsci_basic theme folder, run `npm run test` and make sure everything passes

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
